### PR TITLE
Feature/add house

### PR DESCRIPTION
**Pull request by AlexSJ1990**

### CODE REVIEW

File: src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/transport/TransportDTOs.kt [hunk 1]
```diff
@@ -23,50 +26,59 @@ enum class ApartmentProductSubType {
...
 data class IncompleteQuoteDTO(
         val firstName: String? = null,
         val lastName: String? = null,
         val currentInsurer: String? = null,
         val birthDate: LocalDate?,
         val ssn: String?,
-        val quotingPartner: String?,
-        val incompleteHouseQuoteData: IncompleteHouseQuoteDataDto? = null,
-        val incompleteApartmentQuoteData: IncompleteApartmentQuoteDataDto?
+        @get: Valid val quotingPartner: String?,
```
Comment (johantjellden) [inline src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/transport/TransportDTOs.kt]:
```suggestion
        val quotingPartner: String?,
```

File: src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/transport/TransportDTOs.kt [hunk 2]
```diff
@@ -23,50 +26,59 @@ enum class ApartmentProductSubType {
...
+        val productType: com.hedvig.rapio.quotes.web.dto.ProductType?,
+        val incompleteQuoteData: IncompleteQuoteRequestData
 )
 
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXTERNAL_PROPERTY, property = "productType")
```
Comment (johantjellden) [inline src/main/kotlin/com/hedvig/rapio/externalservices/underwriter/transport/TransportDTOs.kt]: Maybe we should change this to be an internal property.

File: src/main/kotlin/com/hedvig/rapio/quotes/QuotesController.kt [hunk 3]
```diff
@@ -51,50 +61,33 @@ class QuotesController @Autowired constructor(
...
 
     private fun validatePersonalNumber(request: QuoteRequestDTO): Option<QuoteRequestDTO> {
-        return IdNumberValidator.validate(request.quoteData.personalNumber)
-                .map { QuoteRequestDTO.quoteData.personalNumber.set(request, it.idno) }
+        return when (request.quoteData) {
+            is ApartmentQuoteRequestData -> IdNumberValidator.validate(request.quoteData.personalNumber)
+                    .map { request.copy(quoteData = request.quoteData.copy(personalNumber = it.idno)) }
+            is HouseQuoteRequestData -> IdNumberValidator.validate(request.quoteData.personalNumber)
+                    .map { request.copy(quoteData = request.quoteData.copy(personalNumber = it.idno)) }
+        }
```
Comment (johantjellden) [inline src/main/kotlin/com/hedvig/rapio/quotes/QuotesController.kt]: If unused remove this method

File: src/main/kotlin/com/hedvig/rapio/quotes/web/dto/QuoteRequestDTO.kt [hunk 4]
```diff
@@ -1,46 +1,90 @@
...
+    @get:NotBlank val city: String,
+
+    @get:Min(1) @Max(1000) val livingSpace: Int,
+
+    @get:Size(min=12, max=12) val personalNumber: String,
+
+    @get:Min(1) @Max(100)
+    val householdSize: Int,
+
+    @get:NotBlank val ancillaryArea: Int,
```
Comment (johantjellden) [inline src/main/kotlin/com/hedvig/rapio/quotes/web/dto/QuoteRequestDTO.kt]: NotBlank is not needed on Int.

File: src/main/kotlin/com/hedvig/rapio/quotes/QuoteServiceImpl.kt [hunk 5]
```diff
@@ -4,64 +4,89 @@ import arrow.core.Either
...
+        val requestQuoteData = when(quoteData) {
+            is ApartmentQuoteRequestData -> {
+                IncompleteApartmentQuoteDataDto(
+                    street = quoteData.street,
+                    zipCode = quoteData.zipCode,
+                    city = quoteData.city,
+                    livingSpace = quoteData.livingSpace,
+                    householdSize = quoteData.householdSize,
+                    floor = 0,
+                    subType = if (quoteData.productSubType == ProductSubType.RENT) ApartmentProductSubType.RENT else ApartmentProductSubType.BRF
```
Comment (palmenhq) [inline src/main/kotlin/com/hedvig/rapio/quotes/QuoteServiceImpl.kt]: What about students? Maybe we don’t care for now?

File: src/main/kotlin/com/hedvig/rapio/quotes/web/dto/QuoteRequestDTO.kt [hunk 6]
```diff
@@ -1,46 +1,85 @@
...
+data class HouseQuoteRequestData(
+    @get:NotBlank val street: String,
+
+    @get:Digits(integer = 5, fraction = 0)
+    @get:Size(min=5, max=5)
+    @get:NotBlank val zipCode: String,
+
+    @get:NotBlank val city: String,
+
+    @get:Min(1) @Max(1000) val livingSpace: Int,
```
Comment (palmenhq) [inline src/main/kotlin/com/hedvig/rapio/quotes/web/dto/QuoteRequestDTO.kt]: Any particular reason we’re mixing `@get:*` annotations with `@*`?

File: src/main/kotlin/com/hedvig/rapio/quotes/QuoteServiceImpl.kt (hunk 5 above)
Comment (AlexSJ1990) [reply src/main/kotlin/com/hedvig/rapio/quotes/QuoteServiceImpl.kt]: we aren't offering student cover via rapio for now

**Pull request merged**